fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerSimpleTest.java

```diff
@@ -1016,4 +1016,40 @@ public void testMarkReadableClearsBuiltRanges() {
             context.commit(); // fake commit, happy compiler
         }
     }
+
+    @Test
+    public void testTimeLimit() {
+        List<TestRecords1Proto.MySimpleRecord> records = LongStream.range(0, 200).mapToObj(val ->
+                TestRecords1Proto.MySimpleRecord.newBuilder().setRecNo(val).setNumValue2((int)val + 1).build()
+        ).collect(Collectors.toList());
+        Index index = new Index("newIndex", field("num_value_2").ungrouped(), IndexTypes.SUM);
+        IndexAggregateFunction aggregateFunction = new IndexAggregateFunction(FunctionNames.SUM, index.getRootExpression(), index.getName());
+        List<String> indexTypes = Collections.singletonList("MySimpleRecord");
+        FDBRecordStoreTestBase.RecordMetaDataHook hook = metaDataBuilder -> metaDataBuilder.addIndex("MySimpleRecord", index);
+
+        openSimpleMetaData();
+        try (FDBRecordContext context = openContext()) {
+            records.forEach(recordStore::saveRecord);
+            context.commit();
+        }
+
+        openSimpleMetaData(hook);
+        try (OnlineIndexer indexer = OnlineIndexer.newBuilder()
+                .setDatabase(fdb).setMetaData(metaData).setIndex(index).setSubspace(subspace)
+                .setTimeLimitMilliseconds(1)
+                .setLimit(20)
+                .setConfigLoader(old -> {
+                    // Ensure that time limit is exceeded
+                    try {
+                        Thread.sleep(2);
+                    } catch (InterruptedException e) {
+                        e.printStackTrace();
```

Comment:
Instead of printing the stack trace, it might make sense instead to call `fail` with something to indicate that the test was interrupted.

Reply:
Changed.